pkg/operator/controller/ingress/deployment.go

```diff
@@ -568,12 +572,25 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 	}
 
 	dynamicConfigOverride := unsupportedConfigOverrides.DynamicConfigManager
-	if v, err := strconv.ParseBool(dynamicConfigOverride); err == nil && v {
+	if v, err := strconv.ParseBool(dynamicConfigOverride); (err == nil && v) || (err != nil && ingressControllerDCMEnabled) {
```

Comment:
This condition is a little tricky.  What do you think about simplifying it by using an intermediate variable?
```suggestion
	enableDCM := false
	if ingressControllerDCMEnabled {
		enableDCM = true
	}
	dynamicConfigOverride := unsupportedConfigOverrides.DynamicConfigManager
	if v, err := strconv.ParseBool(dynamicConfigOverride); err == nil {
		enableDCM = v
	}
	if enableDCM {
```

Reply:
Done.